crypto/BRCryptoWalletManagerClient.c

```diff
@@ -851,41 +874,41 @@ cwmWalletManagerEventAsETH (BREthereumClientContext context,
-                                                      cryptoWalletManagerTake (cwm),
-                                                      cwmEvent);
+        case EWM_EVENT_CHANGED:
+            // If the newState is `syncing` we want a syncStarted event
+            if (EWM_STATE_SYNCING == event.u.changed.newState) {
+                assert (EWM_STATE_CONNECTED == event.u.changed.oldState);
+                cwm->listener.walletManagerEventCallback (cwm->listener.context,
+                                                          cryptoWalletManagerTake(cwm),
+                                                          (BRCryptoWalletManagerEvent) {
+                                                              CRYPTO_WALLET_MANAGER_EVENT_SYNC_STARTED
+                                                          });
+            }
+
+            // If the oldState is `syncing` we want a syncEnded event
+            if (EWM_STATE_SYNCING == event.u.changed.oldState) {
+                assert (EWM_STATE_CONNECTED == event.u.changed.newState);
+                cwm->listener.walletManagerEventCallback (cwm->listener.context,
+                                                          cryptoWalletManagerTake(cwm),
+                                                          (BRCryptoWalletManagerEvent) {
+                                                              CRYPTO_WALLET_MANAGER_EVENT_SYNC_STOPPED
+                                                          });
+            }
 
             cwmEvent = (BRCryptoWalletManagerEvent) {
                 CRYPTO_WALLET_MANAGER_EVENT_CHANGED,
-                { .state = { cwm->state, CRYPTO_WALLET_MANAGER_STATE_SYNCING }}
-            };
-            cryptoWalletManagerSetState (cwm, CRYPTO_WALLET_MANAGER_STATE_SYNCING);
-            break;
+                { .state = {
+                    cwmStateFromETH (event.u.changed.oldState),
+                    cwmStateFromETH (event.u.changed.newState)
+                }}};
 
-        case EWM_EVENT_SYNC_CONTINUES:
-            cwmEvent = (BRCryptoWalletManagerEvent) {
-                CRYPTO_WALLET_MANAGER_EVENT_SYNC_CONTINUES,
-                { .sync = { 50 }}
-            };
             break;
 
-        case EWM_EVENT_SYNC_STOPPED:
-            cwmEvent = (BRCryptoWalletManagerEvent) {
-                CRYPTO_WALLET_MANAGER_EVENT_SYNC_STOPPED
-            };
-            cwm->listener.walletManagerEventCallback (cwm->listener.context,
-                                                      cryptoWalletManagerTake (cwm),
-                                                      cwmEvent);
-
+        case EWM_EVENT_SYNC_PROGRESS:
             cwmEvent = (BRCryptoWalletManagerEvent) {
-                CRYPTO_WALLET_MANAGER_EVENT_CHANGED,
-                { .state = { cwm->state, CRYPTO_WALLET_MANAGER_STATE_CONNECTED }}
+                CRYPTO_WALLET_MANAGER_EVENT_SYNC_CONTINUES,
+                { .sync = { round (event.u.syncProgress.percent) }}
```

Comment:
I shed a tear for the lost precision.

Reply:
We could do with an addition to `support/<the-most-basis>.h` of `typedef <a number> BRSyncProgressPercentage` - given that we have X (X >=3) current representations.